ukat/mapping/phase_contrast.py

```diff
@@ -0,0 +1,221 @@
+        """
+        with open(path, 'w', newline='') as f:
+            w = csv.DictWriter(f, self.stats_table.keys())
+            w.writeheader()
+            for index in range(self.shape[-1]):
+                temp_dict = {}
+                for key in self.stats_table.keys():
+                    temp_dict[key] = self.stats_table[key][index]
+                w.writerow(temp_dict)
+        del temp_dict
+
+    def to_nifti(self, output_directory=os.getcwd(), base_file_name='Output',
+                 maps='all'):
+        """Exports the velocity array and the renal artery mask to NIFTI.
+
+        Parameters
+        ----------
+        output_directory : string, optional
+            Path to the folder where the NIFTI files will be saved.
+        base_file_name : string, optional
+            Filename of the resulting NIFTI. This code appends the extension.
+            Eg., base_file_name = 'Output' will result in 'Output.nii.gz'.
+        maps : list or 'all', optional
+            List of maps to save to NIFTI. This should either the string "all"
+            or a list of maps from ["phase_array", "mask", "velocity_array"].
+        """
+        os.makedirs(output_directory, exist_ok=True)
+        base_path = os.path.join(output_directory, base_file_name)
+        if maps == 'all' or maps == ['all']:
+            maps = ["velocity_array", "mask"]
+        if isinstance(maps, list):
+            for result in maps:
+                if result == 'velocity_array' or result == 'velocity array':
+                    velocity_nifti = nib.Nifti1Image(self.velocity_array,
+                                                     affine=self.affine)
+                    nib.save(velocity_nifti, base_path +
+                             '_velocity_array.nii.gz')
+                elif result == 'mask':
+                    mask_nifti = nib.Nifti1Image(self.mask.astype(int),
+                                                 affine=self.affine)
+                    nib.save(mask_nifti, base_path + '_mask.nii.gz')
+        else:
+            raise ValueError('No NIFTI file saved. The variable "maps" '
+                             'should be "all" or a list of maps from '
+                             '"["velocity_array", "mask"]".')
+
+
+def convert_to_velocity(image_array, velocity_encoding):
+    """
+    Calculate the velocity array from the given input phase image and
+    velocity encoding value.
+
+    Parameters
+    ----------
+    image_array: np.ndarray
+        A 3D array containing the phase images of the phase contrast sequence.
+    velocity_encoding : float
+        The value of the velocity encoding in cm/s.
+
+    Returns
+    -------
+    velocity_array: np.ndarray
+        A 3D array containing the velocity images of the phase contrast
+        sequence, i.e. the dimensions of the array are [x, y, p], where p
+        corresponds to the phase (or trigger delay).
+    """
+    # velocity_encoding_expected =~ 20cm/s
```

Comment:
Is this left here by mistake?

Reply:
It was a comment to help me and guide me in debugging. Removed now, but this method was extended to deal with GE scans as well.